enhancements/ingress/route-subdomain.md

```diff
@@ -0,0 +1,593 @@
+
+By implementing the `spec.subdomain` field, it will be possible for a Route to
+have one or more host names determined by the router deployment that exposes the
+Route, thus simplifying sharding and other use-cases.
+
+### Goals
+
+* An application administrator can create a Route that is exposed by one or more
+  IngressControllers, and each of the IngressControllers will expose the Route
+  under the IngressController's domain.
+* An application administrator can create a Route with a so-called "vanity"
+  (custom) subdomain without needing to specify a domain.
+* When a user creates a Route with a nonempty value for `spec.subdomain`, the
+  API does not set a value for the Route's `spec.host`.
+* When an IngressController admits a Route with an empty value for `spec.host`
+  and a nonempty value for `spec.subdomain`, the IngressController exposes the
+  Route using a host name comprising the Route's subdomain and the
+  IngressController's domain.
+
+### Non-Goals
+
+* This enhancement does not configure labels or label selectors for sharding.
+* This enhancement does not add any new API fields.
+* This enhancement does not provide any templating mechanism for host names.
+* This enhancement does not allow one Route to have multiple host names for the
+  same router deployment.
+
+## Proposal
+
+Years before this enhancement was written, the `spec.subdomain` field was added
+to the Route API but left unimplemented.  This enhancement implements support
+for `spec.subdomain` in the OpenShift API server and OpenShift router so that so
+that users can specify `spec.subdomain` instead of specifying `spec.host` to get
+a Route that assumes the particular domain of any and all IngressControllers
+that expose it.
+
+Specifically, the OpenShift API server is modified not to set a default value
+for `spec.host` if `spec.subdomain` is specified, and the ingress operator is
+modified to configure router deployments to use the IngressController's
+`status.domain` as the router deployment's domain.
+
+### Validation
+
+The OpenShift API server is modified to validate that the `spec.subdomain` field
+is a valid DNS subdomain as defined in RFC 1123: The value must not exceed 253
+characters and must comprise a period-delimited sequence of labels where each
+label does not exceed 63 characters and does begin with an alphanumeric
+character and contain only alphanumeric characters, possibly with non-initial
+and non-final dashes.  Previously, the OpenShift API server performed no
+validation on the field; however, because the field was unused before this
+enhancement, the stricter validation can be justified as not breaking any
+previously supported use-case; cf. "Risks and Mitigations" below.
+
+The OpenShift API server is also modified not to set a default value for
+`spec.host` if `spec.subdomain` has a nonempty value.  This change enables each
+router deployment to set a host, which is reported in the Route's
+`status.ingress[*].host` field.
```

Comment:
Not quite clear which host the router will set. Or you mean the `status.ingress.host`?

Reply:
A router named "foo" will set the route's `status.ingress["foo"].host` field.  If it will help, I can add the previous sentence to the enhancement document.  Is the `[*]` syntax confusing, is the wording confusing, or something else?

Reply:
Now it's clear, I believe that before you didn't mention that the router updates the status so it could have been interpreted as if the router sets `.spec.host`.